mongo/integration/client_test.go

```diff
@@ -857,8 +857,6 @@ func TestClientStress(t *testing.T) {
 					SetPoolMonitor(tpm.PoolMonitor).
 					SetMaxPoolSize(maxPoolSize))
 			mt.RunOpts(fmt.Sprintf("maxPoolSize %d", maxPoolSize), maxPoolSizeOpt, func(mt *mtest.T) {
-				mt.Parallel()
```

Comment:
Great find! We should also remove the `t.Parallel()` call at the beginning of `TestClientStress`, which should further reduce unexpected interaction with other tests.